tiledb/sm/serialization/query_plan.cc

```diff
@@ -338,26 +339,22 @@ QueryPlan deserialize_query_plan_response(
 
 void serialize_query_plan_request(
     const Config&, Query&, const SerializationType, Buffer&) {
-  throw Status_SerializationError(
-      "Cannot serialize; serialization not enabled.");
+  throw GenericException("Cannot serialize; serialization not enabled.");
```

Comment:
The reason this is a ```GenericException``` and not a ```SerializationStatusException``` is because the ```SerializationStatusException``` class is excluded from the preprocessor when building with serialization off. So we raise a GenericException which is defined in  ```util.h```

Reply:
We should define `class SerializationDisabledException` to use in this case.

Reply:
I defined ```QueryPlanSerializationDisabledException``` so that we are more precise and follow what happens in other classes like ```Array``` as you mentioned